admission_control/admission_control_proto/src/lib.rs

```diff
@@ -76,52 +74,51 @@ pub struct SubmitTransactionResponse {
     pub validator_id: Vec<u8>,
 }
 
-impl IntoProto for SubmitTransactionResponse {
-    type ProtoType = crate::proto::admission_control::SubmitTransactionResponse;
-
-    fn into_proto(self) -> Self::ProtoType {
-        let mut proto = Self::ProtoType::new();
-        if let Some(ac_st) = self.ac_status {
-            proto.set_ac_status(ac_st.into_proto());
-        } else if let Some(mem_err) = self.mempool_error {
-            proto.set_mempool_status(mem_err.into_proto());
-        } else if let Some(vm_st) = self.vm_error {
-            proto.set_vm_status(vm_st.into_proto());
-        } else {
-            error!("No status is available in SubmitTransactionResponse!");
-        }
-        proto.set_validator_id(self.validator_id);
-        proto
-    }
-}
+impl TryFrom<crate::proto::admission_control::SubmitTransactionResponse>
+    for SubmitTransactionResponse
+{
+    type Error = Error;
 
-impl FromProto for SubmitTransactionResponse {
-    type ProtoType = crate::proto::admission_control::SubmitTransactionResponse;
+    fn try_from(proto: crate::proto::admission_control::SubmitTransactionResponse) -> Result<Self> {
+        use crate::proto::admission_control::submit_transaction_response::Status::*;
```

Comment:
why this `use` inside the function and not at the top?

Reply:
It only brings those items into scope for that particular scope. I was trying to limit the damage I was doing and avoid brining too much into scope :)